src/nbl/ext/MitsubaLoader/ParserUtil.cpp

```diff
@@ -44,26 +43,27 @@ void ParserManager::elementHandlerEnd(void* _data, const char* _el)
 
 
 
-bool ParserManager::parse(io::IReadFile* _file)
+bool ParserManager::parse(system::IFile* _file, const system::logger_opt_ptr& _logger)
 {
 	XML_Parser parser = XML_ParserCreate(nullptr);
 	if (!parser)
 	{
-		os::Printer::log("Could not create XML Parser!", ELL_ERROR);
+		_logger.log("Could not create XML Parser!", system::ILogger::E_LOG_LEVEL::ELL_ERROR);
 		return false;
 	}
 
 	XML_SetElementHandler(parser, elementHandlerStart, elementHandlerEnd);
 
 	//from now data (instance of ParserData struct) will be visible to expat handlers
-	Context ctx = {this,parser,io::IFileSystem::getFileDir(_file->getFileName())+"/"};
+	Context ctx = {this,parser,_file->getFileName().parent_path()/""};
```

Comment:
the first XML should parse with parent path/CWD taken from `SAssetLoadContext`
https://github.com/Devsh-Graphics-Programming/Nabla/blob/androidExPort/include/nbl/asset/interchange/IAssetLoader.h#L129

later/deeper paths you can get directly from the `_file`

I suggest that `ParserManager::parse` take the CWD as an argument after the file